src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ItemMapper.java

```diff
@@ -26,11 +29,13 @@
 class ItemMapper {
 
     private final FieldMapper fieldMapper;
-    private final TableMapping tableMapping;
+    private final Map<String, List<FieldMapping>> virtualToPhysicalFieldMappings;
+    private final Map<String, List<FieldMapping>> physicalToVirtualFieldMappings;
 
-    ItemMapper(TableMapping tableMapping, FieldMapper fieldMapper) {
+    ItemMapper(FieldMapper fieldMapper, Map<String, List<FieldMapping>> virtualToPhysicalFieldMappings) {
         this.fieldMapper = fieldMapper;
-        this.tableMapping = tableMapping;
+        this.virtualToPhysicalFieldMappings = virtualToPhysicalFieldMappings;
+        this.physicalToVirtualFieldMappings = invertMapping(virtualToPhysicalFieldMappings);
```

Comment:
I also moved the inversion of the virtual to physical mapping to this class instead of doing it in the `TableMapping`.